<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- Control options -->
    <arg name="leader" default="$(optenv GOPIGO_LEADER false)"/>

    <group ns="gopigo">
        <node pkg="gopigo" name="gopigo_control" type="gopigo_control_node.py" output="screen">
            <param name="distance_update_freq" value="10"/>
            <param name="odometry_update_freq" value="30"/>
        </node>

        <node if="$(arg leader)" pkg="topic_tools" name="leader_req" type="relay" args="reaction command_req"/>
        <node unless="$(arg leader)" pkg="topic_tools" name="follower" type="relay" args="command_req cmd_vel"/>
        <node if="$(arg leader)" pkg="topic_tools" name="leader_vel" type="relay" args="reaction cmd_vel"/>
    </group>
</launch>
